File: server/models/Problem.js
const mongoose = require("mongoose");

const testcaseSchema = new mongoose.Schema({
  input: String,
  output: String,
});

const problemSchema = new mongoose.Schema({
  title: String,
  description: String,
  timeLimit: Number, // ms
  memoryLimit: Number, // KB

  sampleTestcases: [testcaseSchema], // visible
  hiddenTestcases: [testcaseSchema], // used for judging
});

module.exports = mongoose.model("Problem", problemSchema);

File: judge/src/main/java/com/codejudge/judge/model/Problem.java
package com.codejudge.judge.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "problems")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Problem {

    @Id
    private String id;

    private String title;

    private String description;

    private int timeLimit;     // in milliseconds

    private int memoryLimit;   // in MB

    private List<TestCase> sampleTestcases;

    private List<TestCase> hiddenTestcases;

    private String createdBy;  // username (admin)
}